### ISSUE TITLE
Same SDK version, different contents

### ISSUE THREAD

**Issue by natemcmaster** [dotnet/cli#9143]
For the last few rounds of stabilization, we have repeatedly run into issues with ProdCon and the CLI. Because the CLI uses commit height to determine its version, **multiple builds of the CLI will have the same filenames**. This makes it difficult for partner teams to ensure that the correct bits are used and tested.

For example, there are at least two builds of 2.1.300-rc1-008662, one containing MSBuild 15.7.169, the other with MSBuild 15.7.172. Due to the way Azure CDN works, downloading from the CDN (http://dotnetcli.azureedge.net/dotnet/Sdk/2.1.300-rc1-008662/dotnet-sdk-2.1.300-rc1-008662-win-x64.zip) may give you different builds, dependency which network your PC runs on.

**Comment by livarcocc** [comment]
This is not a bug of the SDK, but rather of prodcon builds overwriting builds of the CLI, the same issue will happen for the shared runtime bits as well (and asp.net bits) since they are all behind the same blob storage and CDN.

We should not overwrite builds. The CLI has mechanisms in place to allow the overwriting of the build number.

This is challenging for us because we have multiple legs that are not coordinated by one specific leg, so, we need to be able to re-queue a leg (if it fails) and have predictable results as part of that.

We have needed this in the past when needing to get builds out at 2, 3, 4 in the morning and I am not prepared to give that up without an alternative.

**Comment by muratg** [comment]
@livarcocc are you saying the fix to this would be in prodcon?  In that case, should it be tracked elsewhere?

**Comment by livarcocc** [comment]
No. I agree with Nate that we should be able to have different versions for the rebuilds and that's a change in the CLI. However, we need to be able to do that and to re-spin legs in the same build number when necessary.

It is our work to figure that out.

However, in general, we should not rebuild things in place. It always leads to issues.

**Comment by natemcmaster** [comment]
Can PipeBuild be used to provide consistent build numbers between the various legs of the build?

**Comment by natemcmaster** [comment]
Ping. This burned us again. Can we revisit instead of backlogging things that frequently bring down the build?
